wordpress-mobile/WordPress-Android
# Sync issues: Post conflict handling

Fixes #20531

This PR contains the modifications needed for the new post conflict handling. Specs pcdRpT-5ID-p2#comment-9638.

Note: Existing logic remains unchanged for self-hosted sites; as they do not return conflict response codes.

Merge Instructions
- Ensure FluxC PR has been merged
- Update FluxC hash using the CI merge to trunk hash
- Merge as normal
-----

## To Test:
- Install the Jetpack and WordPress apps from this PR

### Test Post version forced write on top of web (existing logic)
- Login with a site that has posts
- Navigate to Me > Debug Settings
- Disable the `sync_publishing` flag and restart the app
- Navigate to back to My Site > Post
- Tap the publish tab and then tap a post to open the edit view
- PAUSE - GO TO YOUR LAPTOP
- Open a browser and navigate to the same WordPress site
- Open the same post and make an edit (remember what you changed)
- Tap the "update" button to save it and back out of the editor
- GO BACK TO YOUR DEVICE
- Make a change to the post you have in the edit view
- Tap "Update" to save the post to the server
- ✅ Verify the post did update (snackbar)
- GO BACK TO THE LAPTOP
- Tap view on the same post you just edited via the app
- ✅ Verify the changes you made on the device have overwritten the changes you made on your laptop
- Repeat this test for Jetpack and Wordpress

### Test Post version did not write on top of web (new logic)
- Navigate to Me > Debug Settings
- Enabled the `sync_publishing` flag and restart the app 
- Navigate to back to My Site > Post
- Tap the publish tab and then tap a post to open the edit view
- PAUSE - GO TO YOUR LAPTOP
- Open a browser and navigate to the same WordPress site
- Open the same post and make an edit (remember what you changed)
- Tap the "update" button to save it and back out of the editor
- GO BACK TO YOUR DEVICE
- Make a change to the post you have in the edit view
- Tap "Update" to save the post to the server
- ✅ Verify the post did not update and the post list shows "Version Conflict" within the post row
- GO BACK TO THE LAPTOP
- Tap view on the same post you just edited via the app
- ✅ Verify the changes you made on the device did not overwrite the changes you made on your laptop
- Repeat this test for Jetpack and Wordpress

### Test Conflict Resolution dialog is shown
- Continue from the last step of test "Post version did not write on top of web (new logic)"
- On the post list, tap on the post that shows "Version Conflict"
- ✅ Verify the conflict resolution dialog is shown
- Tap local version
- ✅ Verify the local version has been applied
- Repeat the scenario in  test "Post version did not write on top of web (new logic)" to get into a Version Conflict again. Then repeat this test and tap web version in the conflict resolution dialog.

-----

## Regression Notes

1. Potential unintended areas of impact
The post is not updated properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Updated existing unit tests.
Added `PostConflictResolverTest` and `SyncPublishingFeatureUtilsTest`

3. What automated tests I added (or what prevented me from doing so)
Relied upon existing automated post tests

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): No UI components were changed with this PR
